### PR TITLE
Fix Mutex.try_lock in systhreads.

### PR DESCRIPTION
**Pull request by abbysmal**
See #425
In what appears to be a terrible mistake, it seems I inverted the result of `Mutex.try_lock` during the systhreads port to multicore.
Indeed it would returns `false` on successfully locking, and `true` on being unable to lock.
Sorry for this mistake, as @ctk21 mentioned I think it is for the better if we actually add `try_lock` to the testsuite, in order to avoid such small mistakes.
I extracted the test program provided by @toots to use as our new testcase for `try_lock`.

### CODE REVIEW

File: otherlibs/systhreads/st_stubs.c [hunk 1]
```diff
@@ -554,7 +554,7 @@ CAMLprim value caml_mutex_try_lock(value wrapper)           /* ML */
 {
   st_mutex mut = Mutex_val(wrapper);
   int retcode = st_mutex_trylock(mut);
-  if (retcode == 1) return Val_false;
+  if (retcode == 0) return Val_false;
```
Comment (ctk21) [inline otherlibs/systhreads/st_stubs.c]: Is it better do something a bit clearer and use a define for the return code like:
https://github.com/ocaml/ocaml/blob/4.10/otherlibs/systhreads/st_stubs.c#L846

**Review by ctk21 — COMMENTED** [review]
Can we also add the positive testcase for trylock?

That is, something along the lines of:
```ocaml
let () =
  let m = Mutex.create () in
  assert (Mutex.try_lock m);
  Mutex.unlock m
```

**Comment by abbysmal** [comment]
Thank you for your comments, I just pushed a positive testcase for trylock.
I also updated and reused the defines for trylock return values to match more how trunk handles trylock's returned values.

**Comment by toots** [comment]
Great, can't wait to test the fixed compiler!

**Comment by ctk21** [comment]
LGTM - merging!

**Pull request merged**